gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java

```diff
@@ -96,30 +101,53 @@ public GitCoreRemoteBranch getRemoteBranch(String branchName) throws GitCoreExce
           LOG.debug(() -> "* ${branch.getName()}");
           return branch;
         })
-        .map(ref -> new GitCoreLocalBranch(/* repo */ this,
-            ref.getName().replace(GitCoreLocalBranch.BRANCHES_PATH, /* replacement */ "")))
+        .map(ref -> {
+          String shortBranchName = ref.getName().replace(GitCoreLocalBranch.BRANCHES_PATH, /* replacement */ "");
+          String remoteName = deriveRemoteName(shortBranchName);
+          return new GitCoreLocalBranch(/* repo */ this, shortBranchName, remoteName,
+              Try.of(() -> getRemoteBranch(shortBranchName, remoteName).getOrNull()).getOrNull());
+        })
         .collect(List.collector());
   }
 
   @Override
-  public List<IGitCoreRemoteBranch> getRemoteBranches() throws GitCoreException {
-    LOG.debug(() -> "Entering: repository = ${mainDirectoryPath} (${gitDirectoryPath})");
-    LOG.debug("List of remote branches:");
-    return Try.of(() -> getJgitGit().branchList().setListMode(ListBranchCommand.ListMode.REMOTE).call())
+  public List<IGitCoreRemoteBranch> getRemoteBranches(String remoteName) throws GitCoreException {
+    LOG.debug(() -> "Entering: remoteName = ${remoteName}, repository = ${mainDirectoryPath} (${gitDirectoryPath})");
+    LOG.debug("List of remote branches of '${remoteName}':");
+    return Try.of(() -> getJgitRepo().getRefDatabase().getRefsByPrefix(GitCoreRemoteBranch.BRANCHES_PATH + remoteName + "/"))
         .getOrElseThrow(e -> new GitCoreException("Error while getting list of remote branches", e))
         .stream()
```

Comment:
While you're here: get rid of `stream()`s in this file, apply `List.ofAll` on the `getOrElseThrow`'s result and use Vavr List methods (including `forEach` for logging etc.)

Reply:
Ok, but for logging I must use `map` because `forEach` is terminal statement. I also can split this method chain, first part assign to variable and then apply to it `forEach` and next do the rest of chain on this variable. WDYT?

Reply:
Oh ok, I forget about `forEach` being terminal, you brought up this point before... so do you judgement, can stay as is as well